packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx

```diff
@@ -168,6 +170,7 @@ export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
       }
     };
     // set initial state for textOnlyChat and isACSChat
+    // eslint-disable-next-line @internal/custom-rules/no-getstate
     updateChatState(adapter.getState());
```

Comment:
It looks like this can be updated to use `useSelector` and `useEffect`. `adapter.getState()` usage can be removed then

Reply:
Great point, I wasn't certain what this was for, looking at it again yeah we should be able to remove this whole useEffect and onStateChange subscription and have:
```ts
const textOnlyChat = useSelector(state => !!state..thread.properties?.messagingPolicy?.textOnlyChat);
const createdBy = useSelector(state => state.thread.properties?.createdBy)
const isACSChat = !isMicrosoftTeamsUserIdentifier(createdBy)
```

Reply:
Sounds great!
Do you think it will be useful to use `useMemo` here for `isACSChat`?

Reply:
Booleans probably don't need to be memoized
https://github.com/Azure/communication-ui-library/pull/5239/files/7081a38f897d1372395e08952c9c45bfb6cb7582#r1788032458

Reply:
Updated